Map Generation Fixes and Tweaks

r3versi opened the pull request:
* Renamed **Continents** to **Two Continents**
* Created **Four Continents** map type
It spawns a vertical and a horizontal ocean to split the map in 4 major continents. Works properly on all sizes and with world wrap
![Screenshot from 2021-06-07 17-34-07](https://user-images.githubusercontent.com/17114100/121048390-0d3d5d00-c7b7-11eb-827e-473d5fb21ee2.png)
![Screenshot from 2021-06-07 17-34-43](https://user-images.githubusercontent.com/17114100/121048399-0e6e8a00-c7b7-11eb-899a-00affc8148c0.png)
* Fixed Pangea extending to boundaries in rectangular maps (Fix #3057)
OLD:
![Screenshot from 2021-06-07 17-40-27](https://user-images.githubusercontent.com/17114100/121048985-96549400-c7b7-11eb-83ef-36f90e0ed416.png)
NEW:
![Screenshot from 2021-06-07 17-35-18](https://user-images.githubusercontent.com/17114100/121048403-0f072080-c7b7-11eb-943e-e429cebf2fc9.png)
* Unified Pangea for rectangular and hexagonal maps
* Fixed Default map type bias on Rectangular maps. Also removed "pangea effect" on Default: land can spawn with the same probability in the whole map.
OLD:
![Screenshot from 2021-06-07 17-41-04](https://user-images.githubusercontent.com/17114100/121049189-c865f600-c7b7-11eb-9f18-e5419199963c.png)
NEW:
![Screenshot from 2021-06-07 17-35-51](https://user-images.githubusercontent.com/17114100/121048407-10384d80-c7b7-11eb-9099-238318c147ea.png)

yairm210 commented:
A Looks good!
B. Please add template.properties translation entries for new map types (and remove for old)
C. What's the compatibility with word wrap maps?

r3versi commented:
> A Looks good!
> B. Please add template.properties translation entries for new map types (and remove for old)
> C. What's the compatibility with word wrap maps?


B. Done. I renamed Four Continents to Four Corners as suggested by @ravignir and reverted the change from Continents to Two Continents

C. Fully compatible with world wrap

SomeTroglodyte commented:
:+1: ... Maybe Four Corners needs a slight tweak to make a few more desert tiles? Those look awfully rare ...
